### PR TITLE
Jones/users and awards

### PR DESCRIPTION
**Pull request by taylor-jones**
Branched from Reitz/log work, to integrate with logging:

1. Implements controllers and services for CRUD operations with UserAccount and Award.
  > Both are integrated with the LogService.

2. Added equals(), hashcode(), and toSring() methods in each of the models. I'm still largely unfamiliar with this practice, but I believe this is better allowing us to make comparisons between these objects during CRUD operations. 
  > For instance, in the Log table, if you look at the ```changed_from``` and ```changed_to``` columns for records 16-18 and 22-24, you'll notice that it's not very indicative of what's actually changing. After adding the toString() method to the models, the ```changed_from``` and ```changed_to``` columns show more descriptive content (see rows 117-119 and 123-125 as examples).

3. With that said, the ```changed_from``` and ```changed_to``` columns were still showing data that wasn't actually being changed. For example, in rows 141 and 142 of the Log table, you'll notice there isn't actually a change between the ```changed_from``` and ```changed_to``` columns. 
> In an effort to remedy this, I made a small change to the LogService.entityDiff() function comparison that compares the .toString() values. I'm not sure how reasonable this solution is, but it did prevent unchanged values from showing up in the Log table as updates.

4. I noticed the Log model getters didn't have the `get` keyword before them, which was causing the issue of not showing these columns in the GET response, so I updated the getters.

### CODE REVIEW

File: erp-server/src/main/java/com/ttt/erp/service/LogService.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        startFields[i].setAccessible(true);
+        endFields[i].setAccessible(true);
+
+        String strStart = startFields[i].get(start).toString();
+        String strEnd = startFields[i].get(end).toString();
+
+        if (!strStart.equals(strEnd)) {
+          diffs.add (
+            new PropertyChange(startFields[i].getName(), strStart, strEnd)
+          );
```
Comment (tmrietz) [inline erp-server/src/main/java/com/ttt/erp/service/LogService.java:59]: Nice! This is much better.

**Comment by taylor-jones** [comment]
Have updated with several new endpoints since initial PR

**Review by tmrietz — APPROVED** [review]
Awesome! Nicely done, I’m good with this getting merged.

Edit: Still looking into Logs. I'd still say go ahead and merge since there's so much other stuff we should get merged.

File: erp-server/src/main/java/com/ttt/erp/service/LogService.java (hunk 1 above)
Comment (tmrietz) [reply erp-server/src/main/java/com/ttt/erp/service/LogService.java:59]: I need to spend some more time looking at this actually. Current solution might be tough to report on.

**Pull request merged**